aiohomekit/controller/coap/connection.py

```diff
@@ -0,0 +1,971 @@
+
+import asyncio
+import struct
+import logging
+import tlv8
+import uuid
+
+from aiocoap import Context, Message, resource
+from aiocoap.numbers.codes import Code
+from cryptography.exceptions import InvalidTag
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey, Ed25519PublicKey
+from cryptography.hazmat.primitives.asymmetric.x25519 import X25519PrivateKey, X25519PublicKey
+from cryptography.hazmat.primitives.ciphers.aead import ChaCha20Poly1305
+
+from aiohomekit.crypto.hkdf import hkdf_derive
+from aiohomekit.crypto.srp import SrpClient
+
+
+# globals/enums
+logger = logging.getLogger(__name__)
+
+class HAP_PAIRING_ERRORS:
```

Comment:
There are already constants for these, again inherited from homekit_python. They are not in the most useful format but we should try and consolidated (search for kTLVError_MaxPeers).

Reply:
I'm also trying to use python enums for new things, as its more useful for type checking and code completion.

Reply:
Removed errors & methods as they aren't needed now that crypto is merged.